prow/e2e-kind-suite.sh

```diff
@@ -42,11 +42,17 @@ setup_and_export_git_sha
 function build_kind_images() {
   # Build just the images needed for the tests
   for image in pilot proxyv2 proxy_init app test_policybackend mixer citadel galley sidecar_injector kubectl node-agent-k8s; do
-     make docker.${image}
+    DOCKER_BUILD_VARIANTS="${VARIANT:-default}" make docker.${image}
   done
 	# Archived local images and load it into KinD's docker daemon
 	# Kubernetes in KinD can only access local images from its docker daemon.
 	docker images "${HUB}/*:${TAG}" --format '{{.Repository}}:{{.Tag}}' | xargs -n1 -P16 kind --loglevel debug --name istio-testing load docker-image
+
+	# If a variant is specified, load those images as well.
+	# We should load non-variant images as well for things like `app` which do not use variants
+	if [[ "${VARIANT:-}" != "" ]]; then
```

Comment:
Nit: I think this is more consistent with the style of the rest of this script

```suggestion
	if [[ -n "${VARIANT:-}" ]]; then
```